src/ShoppingCart/Cart.js

```diff
@@ -1,8 +1,58 @@
-import React from "react";
+import React, { useState } from "react";
 import CartModal from "./CartModal";
 import ProductList from "./ProductList";
 
 export default function Cart() {
+  const [cartArray,setCartArray] = useState([]);
+  const addProduct = (product) => {
+    //check the id of product, where it is in the cartArray or not
+    let indexProduct = cartArray.findIndex((item) => item.id === product.id);
+    let updateQuantityArray = [...cartArray];
+    console.log("index of products", indexProduct);
+    if (indexProduct !== -1) {
+      console.log("Product exists already need to update the quantity");
+      console.log(indexProduct);
+      updateQuantityArray[indexProduct].quantity += 1;
+      //find the product is in the array, update the quantity of the product
+      setCartArray(updateQuantityArray);
+    } else {
+      console.log("Product new need to push to the array");
+      //update the object product
+      let updatedProduct = {...product, quantity: 1};
+      updateQuantityArray = [...cartArray, updatedProduct];
+      setCartArray(updateQuantityArray);
+    }
+    //update the array of cart
+  };
+
+//=====================remove the product=============================
+  const removeProduct = (id) =>{
+    console.log(id);
+    console.log(cartArray);
+
+    let updatedProductArray = [...cartArray]
+    let indexProduct = updatedProductArray.findIndex((cartItem) => {
+      return cartItem.id === id;
+    });
+    if (indexProduct !== -1){
+      updatedProductArray.splice(indexProduct,1);
```

Comment:
always check whether the product is available in the array or not. If yes, using splice to remove